eos-updater/apply.c

```diff
@@ -247,6 +276,12 @@ apply_internal (ApplyData     *apply_data,
                local_error->message);
   g_clear_error (&local_error);
 
+  /* Prune any unused objects in the repo. This is a necessary step
+   * after removing old refs. */
+  prune_repo_and_report_statistics (repo,
+                                    OSTREE_REPO_PRUNE_FLAGS_REFS_ONLY,
+                                    cancellable);
```

Comment:
Yeah, I had this since there didn't seem to be anything about pruning in the documentation for `ostree_sysroot_cleanup`, but checking here, it does a prune as you suggest: https://github.com/resin-io/ostree/blob/master/src/libostree/ostree-sysroot-cleanup.c#L474 . I'll dro the commit.

Reply:
Yeah, you are definitely right that the documentation is not clear on that.